ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -2452,9 +2448,16 @@ private Partition loadPartitionInternal(Path loadPath, Table tbl, Map<String, St
         skewedInfo.setSkewedColValueLocationMaps(skewedColValueLocationMaps);
         newCreatedTpart.getSd().setSkewedInfo(skewedInfo);
       }
-      if (!this.getConf().getBoolVar(HiveConf.ConfVars.HIVESTATSAUTOGATHER)) {
+
+      //if hive.stats.autogather = false or resetStatistics = true then
+      //clear partition column stats and set basic stats to false
+      if (!this.getConf().getBoolVar(HiveConf.ConfVars.HIVESTATSAUTOGATHER) || resetStatistics) {
+        LOG.debug(
+            "clear partition column statistics and setting basic stats to false for " + newTPart.getCompleteName());
+        StatsSetupConst.clearColumnStatsState(newTPart.getParameters());
```

Comment:
clearColumnStatsState call is redundant as setBasicStatsState(false) will remove the column stats as well.